modules/explorer/update.go

```diff
@@ -14,36 +11,298 @@ func (e *Explorer) ProcessConsensusChange(cc modules.ConsensusChange) {
+
+			for _, sci := range txn.SiacoinInputs {
+				delete(e.siacoinOutputIDs[sci.ParentID], txid)
+				delete(e.unlockHashes[sci.UnlockConditions.UnlockHash()], txid)
+				e.siacoinInputCount--
+			}
+			for k, sco := range txn.SiacoinOutputs {
+				delete(e.siacoinOutputIDs[txn.SiacoinOutputID(uint64(k))], txid)
+				delete(e.unlockHashes[sco.UnlockHash], txid)
+				e.siacoinOutputCount--
+			}
+			for k, fc := range txn.FileContracts {
+				fcid := txn.FileContractID(uint64(k))
+				delete(e.fileContractIDs[fcid], txid)
+				delete(e.unlockHashes[fc.UnlockHash], txid)
+				for l, sco := range fc.ValidProofOutputs {
+					scoid := fcid.StorageProofOutputID(types.ProofValid, uint64(l))
+					delete(e.siacoinOutputIDs[scoid], txid)
+					delete(e.unlockHashes[sco.UnlockHash], txid)
+				}
+				for l, sco := range fc.MissedProofOutputs {
+					scoid := fcid.StorageProofOutputID(types.ProofMissed, uint64(l))
+					delete(e.siacoinOutputIDs[scoid], txid)
+					delete(e.unlockHashes[sco.UnlockHash], txid)
+				}
+				e.fileContractCount--
 				e.totalContractCost = e.totalContractCost.Sub(fc.Payout)
 				e.totalContractSize = e.totalContractSize.Sub(types.NewCurrency64(fc.FileSize))
 			}
 			for _, fcr := range txn.FileContractRevisions {
-				e.totalContractCount -= 1
+				delete(e.fileContractIDs[fcr.ParentID], txid)
+				delete(e.unlockHashes[fcr.UnlockConditions.UnlockHash()], txid)
+				delete(e.unlockHashes[fcr.NewUnlockHash], txid)
+				for l, sco := range fcr.NewValidProofOutputs {
+					scoid := fcr.ParentID.StorageProofOutputID(types.ProofValid, uint64(l))
+					delete(e.siacoinOutputIDs[scoid], txid)
+					delete(e.unlockHashes[sco.UnlockHash], txid)
+				}
+				for l, sco := range fcr.NewMissedProofOutputs {
+					scoid := fcr.ParentID.StorageProofOutputID(types.ProofMissed, uint64(l))
+					delete(e.siacoinOutputIDs[scoid], txid)
+					delete(e.unlockHashes[sco.UnlockHash], txid)
+				}
+				e.fileContractRevisionCount--
 				e.totalContractSize = e.totalContractSize.Sub(types.NewCurrency64(fcr.NewFileSize))
 				e.totalRevisionVolume = e.totalRevisionVolume.Sub(types.NewCurrency64(fcr.NewFileSize))
 			}
+			for _, sp := range txn.StorageProofs {
+				delete(e.fileContractIDs[sp.ParentID], txid)
+				e.storageProofCount--
+			}
+			for _, sfi := range txn.SiafundInputs {
+				delete(e.siafundOutputIDs[sfi.ParentID], txid)
+				delete(e.unlockHashes[sfi.UnlockConditions.UnlockHash()], txid)
+				delete(e.unlockHashes[sfi.ClaimUnlockHash], txid)
+				e.siafundInputCount--
+			}
+			for k, sfo := range txn.SiafundOutputs {
+				sfoid := txn.SiafundOutputID(uint64(k))
+				delete(e.siafundOutputIDs[sfoid], txid)
+				delete(e.unlockHashes[sfo.UnlockHash], txid)
+				e.siafundOutputCount--
+			}
+			for _ = range txn.MinerFees {
+				e.minerFeeCount--
+			}
+			for _ = range txn.ArbitraryData {
+				e.arbitraryDataCount--
+			}
+			for _ = range txn.TransactionSignatures {
+				e.transactionSignatureCount--
```

Comment:
could just subtract the `len` but I guess this leaves room for doing fancier stuff in the future